### PR TITLE
added basic profile page

### PR DESCRIPTION
**Pull request by mattorib**
Implemented the CoordinatorLayout, RecyclerView, and an extremely basic CardView for profiles. Fully implemented ViewHolder for the cards will be in a later PR.

### CODE REVIEW

**Comment by mattorib** [comment]
screenshot_20161206-000823
screenshot_20161206-000844

**Comment by mattorib** [comment]
screenshot_20161206-165540
screenshot_20161206-165551

File: app/build.gradle.orig [hunk 1]
```diff
@@ -0,0 +1,53 @@
+apply plugin: 'com.android.application'
```
Comment (LarryAsante) [inline app/build.gradle.orig]: This file should not be in here

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Activities/MainActivity.java [hunk 2]
```diff
@@ -116,9 +118,12 @@ private void setUpToolbarToggle() {
 
     private void replaceFrameWithFragment(Fragment fragment){
         FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
-        transaction.replace(R.id.frame, fragment);
+        transaction.replace(R.id.frame_layout, fragment);
         transaction.commit();
     }
+    public void openDrawer(){
+        drawerLayout.openDrawer(GravityCompat.START);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Activities/MainActivity.java]: Nit-picking: Use a static import for this.  This should read `drawerLayout.openDrawer(START);`. See above for the import statement.

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Activities/MainActivity.java [hunk 3]
```diff
@@ -5,12 +5,14 @@
 import android.support.design.widget.NavigationView;
 import android.support.v4.app.Fragment;
 import android.support.v4.app.FragmentTransaction;
+import android.support.v4.view.GravityCompat;
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Activities/MainActivity.java]: Change this to `import static android.support.v4.view.GravityCompat.START;` and move this to the bottom of the import statements

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Adapters/ArticleAdapter.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    public ArticleAdapter(List<RealmStory> stories) {
+        this.stories = stories;
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View v = LayoutInflater.from(parent.getContext())
+                               .inflate(R.layout.viewholder_article, parent, false);
+        ArticleViewHolder vh = new ArticleViewHolder(v);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Adapters/ArticleAdapter.java]: Please use proper variable names

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Adapters/ArticleAdapter.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        }
+    }
+
+    public ArticleAdapter(List<RealmStory> stories) {
+        this.stories = stories;
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View v = LayoutInflater.from(parent.getContext())
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Adapters/ArticleAdapter.java]: Use proper variable naming

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+            public void onClick(View v) {
+                Snackbar.make(v, "Sample article added to Favorites", Snackbar.LENGTH_SHORT).show();
+            }
+        });
+    }
+
+    /** Fills fragment with placeholder content */
+    private void loadPlaceHolderData(){
+        Drawable defaultImage = ContextCompat.getDrawable(getContext(), R.drawable.grinnell_gates);
+        setHeaderText("Sample title");
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java]: Even though this is a dummy string, extract this text into the strings.xml and reference it from there.  Please label it as such in the strings.xml file

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        setContentText("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris feugiat diam sollicitudin est semper, eu porta libero pulvinar. Aenean at lacus rhoncus, pharetra elit ut, ultricies magna.");
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java]: Same here

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+
+/**
+ * Fragment that displays the expanded article content.
+ * Fragment allows for favoriting of the article.
+ */
+
+public class ExpandedArticleFragment extends Fragment implements UserInterface {
+
+    private ImageView mHeaderImage;
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java:35]: I like the consistency in your variable naming! 👍

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater,
+                             ViewGroup container,
+                             Bundle savedInstanceState) {
+        final View expandedArticleFragment = inflater.inflate(R.layout.fragment_extended_article, container, false);
+        mHeaderImage = (ImageView) expandedArticleFragment.findViewById(R.id.header_image);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java]: Our `onCreateView(...)` is getting cluttered. Extract all of the view bindings here into a method, say `public void bindView()` and call that method in `initializeUI()`, above `loadPlaceHolderData()`

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+            @Override
+            public void onClick(View v) {
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        mFavoriteButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Snackbar.make(v, "Sample article added to Favorites", Snackbar.LENGTH_SHORT).show();
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java]: nit: Use static import

Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java]: Extract string into the `strings.xml` file

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+        mArticleToolbar.setNavigationOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        mFavoriteButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java]: I know this was the variable name given on the method's call but please change it to something like `View view`. It's more descriptive

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+                Snackbar.make(v, "Sample article added to Favorites", Snackbar.LENGTH_SHORT).show();
+            }
+        });
+    }
+
+    /** Fills fragment with placeholder content */
+    private void loadPlaceHolderData(){
+        Drawable defaultImage = ContextCompat.getDrawable(getContext(), R.drawable.grinnell_gates);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java]: You could also use a static import for `getDrawable()` as it is a static method of the `ContextCompat` class

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+ * Fragment allows for favoriting of the article.
+ */
+
+public class ExpandedArticleFragment extends Fragment implements UserInterface {
+
+    private ImageView mHeaderImage;
+    private CollapsingToolbarLayout mCollapsingToolbar;
+    private FloatingActionButton mFavoriteButton;
+    private TextView mArticleContent;
+    private Toolbar mArticleToolbar;
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ExpandedArticleFragment.java:39]: Please destroy all of these member variables in Fragment's `onDestroyView()` as it allows the fragment to clean up resources associated with its View. That way, we're not leaking any memory references on the Fragment's destruction.

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ProfileFragment.java [hunk 14]
```diff
@@ -1,20 +1,40 @@
...
+import edu.grinnell.grinnell_publications_android.Models.Realm.RealmStory;
 import edu.grinnell.grinnell_publications_android.R;
+import io.realm.RealmList;
 
 /**
  * {@link Fragment} to show a user's Profile
  * @author Larry Boateng Asante
  */
 public class ProfileFragment extends Fragment implements UserInterface {
+    private RecyclerView mRecycler;
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ProfileFragment.java:34]: Please destroy all of these member variables in Fragment's `onDestroyView()` as it allows the fragment to clean up resources associated with its View. That way, we're not leaking any memory references on the Fragment's destruction.

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ProfileFragment.java [hunk 15]
```diff
@@ -24,12 +44,59 @@ public ProfileFragment() {
     public View onCreateView(LayoutInflater inflater,
                              ViewGroup container,
                              Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_profile, container, false);
+        mRecycler = (RecyclerView) view.findViewById(R.id.articles_recyclerview);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ProfileFragment.java]: Extract to a `bindView()` method and call in `initializeUI()`

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ProfileFragment.java [hunk 16]
```diff
@@ -24,12 +44,59 @@ public ProfileFragment() {
...
+        rv.setAdapter(new ArticleAdapter(retrieveStories()));
+    }
+
+    private List<RealmStory> retrieveStories() {
+        // TODO: implement with network
+        // for now, dummy data
+        RealmPublication snb = new RealmPublication("S&B", 0, null, null, null);
+        RealmList<RealmPublication> pubs = new RealmList<>();
+        pubs.add(snb);
+        RealmAuthor richard = new RealmAuthor("Rick and Morty", null, pubs, null);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ProfileFragment.java]: Please use proper variable names

File: app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ProfileFragment.java [hunk 17]
```diff
@@ -24,12 +44,59 @@ public ProfileFragment() {
...
+        RecyclerView.LayoutManager lm = new LinearLayoutManager(getContext());
+        rv.setLayoutManager(lm);
+
+        rv.setAdapter(new ArticleAdapter(retrieveStories()));
+    }
+
+    private List<RealmStory> retrieveStories() {
+        // TODO: implement with network
+        // for now, dummy data
+        RealmPublication snb = new RealmPublication("S&B", 0, null, null, null);
```
Comment (LarryAsante) [inline app/src/main/java/edu/grinnell/grinnell_publications_android/Fragments/ProfileFragment.java]: Proper variable names

File: app/src/main/res/values/colors.xml [hunk 18]
```diff
@@ -2,10 +2,11 @@
 <resources>
     <color name="colorPrimary">#212121</color>
     <color name="colorPrimaryDark">#000000</color>
-    <color name="colorAccent">#DB6D6A </color>
 
     <color name="black">#000000</color>
     <color name="white">#FFFFFF</color>
     <color name="grey">#DDDDDD</color>
     <color name="scarlet_red">#E1261C</color>
+
```
Comment (LarryAsante) [inline app/src/main/res/values/colors.xml]: Unnecessary whitespace. Take it off

File: app/src/main/res/values/styles.xml [hunk 19]
```diff
@@ -16,11 +16,17 @@
...
         <item name="android:textColorPrimary">@color/black</item>
         <item name="android:activatedBackgroundIndicator">@drawable/drawer_list_selector</item>
     </style>
 
+    <style name="MaterialAnimations" parent="@style/Theme.AppCompat.Light.NoActionBar">
+        <item name="android:windowContentTransitions">true</item>
+    </style>
+
+
+
```
Comment (LarryAsante) [inline app/src/main/res/values/styles.xml]: Unnecessary whitespace. Take it off

**Review by LarryAsante — CHANGES_REQUESTED** [review]
Great job getting it done, Mattori. Please check the comments and make the appropriate changes. Thanks for all your work!

**Comment by mattorib** [comment]
@LarryAsante Please take a close look at the usage of `onDestroyView` and the update to the `initializeUI` interface method (I accidentally typo'd that commit by calling it `initializeView` 😞 )

**Comment by mattorib** [comment]
screenshot_20170120-205204
screenshot_20170120-205219

**Pull request merged**